pyfr/solvers/base/elements.py

```diff
@@ -47,6 +47,9 @@ def __init__(self, basiscls, eles, cfg):
         self.nfpts = basis.nfpts
         self.nfacefpts = basis.nfacefpts
 
+        # Compute metric (Smat, djac on pseudo grid points)
+        self._smat_base()
```

Comment:
I am wondering that even _smat_base() is removed in the constructor, the next line, _gen_pnorm_fpts(), requires smats and eventually calls _smat_base(). The constructor cost will be the same.

Do you think it may be better to remove _get_pnorm_fpts() in the constructor, as well ?